lib/src/kvs/ds.rs

```diff
@@ -237,24 +241,74 @@ impl Datastore {
 		}
 	}
 
-	// Adds entries to the KV store indicating membership information
-	pub async fn register_membership(&self) -> Result<(), Error> {
+	// -----
+	// Cluster helpers here
+	// -----
+	pub async fn cluster_init(&self) -> Result<(), Error> {
+		trace!("First register to prove cluster registration works");
+		let timestamp = self.register_membership().await?;
+		trace!("Healthcheck to perform garbage collection");
+		// node timeout should be configurable, just trying to get this to work
+		// let timeout = Duration::from_secs(1);
+		// let deadline = now - timeout;
+		let watermark = timestamp;
+		self.garbage_collect(watermark).await?;
+		trace!("Second register to prove removal worked and there is no conflict");
+		self.register_membership().await?;
+		Ok(())
+	}
+
+	pub async fn garbage_collect(&self, watermark: Timestamp) -> Result<(), Error> {
 		let mut tx = self.transaction(true, false).await?;
-		tx.set_cl(sql::Uuid::from(*self.id.as_ref())).await?;
-		tx.set_hb(sql::Uuid::from(*self.id.as_ref())).await?;
+		let dead_heartbeats = self.delete_dead_heartbeats(&mut tx, watermark).await?;
+		trace!("Found dead hbs: {:?}", dead_heartbeats);
+		for hb in dead_heartbeats {
+			tx.del_cl(hb.nd).await?;
+			trace!("Deleted node {}", hb.nd);
+		}
 		tx.commit().await?;
 		Ok(())
 	}
```

Comment:
This method only GC heartbeats for a node with the same ID, right? How do you handle a node (uuid) being deleted for good?

Reply:
Nevermind, I reread the PR and I see that the Heartbeat range includes all nodes for a given time range.